DigitalLearningSolutions.Web/ViewModels/TrackingSystem/Delegates/DelegateProgress/EditDelegateCourseAdminFieldFormData.cs

```diff
@@ -0,0 +1,37 @@
+﻿namespace DigitalLearningSolutions.Web.ViewModels.TrackingSystem.Delegates.DelegateProgress
+{
+    using System.ComponentModel.DataAnnotations;
+    using DigitalLearningSolutions.Data.Models.Courses;
+
+    public class EditDelegateCourseAdminFieldFormData
+    {
+        public EditDelegateCourseAdminFieldFormData() { }
+
+        protected EditDelegateCourseAdminFieldFormData(DelegateCourseDetails details, int? returnPage)
+        {
+            DelegateId = details.DelegateCourseInfo.DelegateId;
+            CustomisationId = details.DelegateCourseInfo.CustomisationId;
+            DelegateName = details.DelegateCourseInfo.DelegateFirstName == null
+                ? details.DelegateCourseInfo.DelegateLastName
+                : $"{details.DelegateCourseInfo.DelegateFirstName} {details.DelegateCourseInfo.DelegateLastName}";
+            ReturnPage = returnPage;
+        }
+
+        protected EditDelegateCourseAdminFieldFormData(EditDelegateCourseAdminFieldFormData formData)
+        {
+            Answer = formData.Answer;
+            DelegateId = formData.DelegateId;
+            CustomisationId = formData.CustomisationId;
+            DelegateName = formData.DelegateName;
+            ReturnPage = formData.ReturnPage;
+        }
+
+        [MaxLength(100, ErrorMessage = "Answer must be 100 characters or fewer")]
+        public string? Answer { get; set; }
+
+        public int DelegateId { get; set; }
+        public string? DelegateName { get; set; }
+        public int CustomisationId { get; set; }

```

Comment:
I'm also not sure whether all these belong on the FormData in this case since we need to go and query the DelegateCourseDetails on validation failure anyway.

Reply:
Yep, delegateId and customisationId are used to redirect. I've moved DelegateName to the view model